tests/runInMemory.hs

```diff
@@ -0,0 +1,32 @@
+-- Run tests using the TCP transport.
+
+module Main where
+
+import TEST_SUITE_MODULE (tests)
+
+import Network.Transport.Test (TestTransport(..))
+import Network.Transport.InMemory
+import Test.Framework (defaultMainWithArgs)
+
+import Control.Concurrent (threadDelay)
+import System.Environment (getArgs)
+
+main :: IO ()
+main = do
+    (transport, internals) <- createTransportExposeInternals
+    ts <- tests TestTransport
+      { testTransport = transport
+      , testBreakConnection = \addr1 addr2 -> do breakConnection internals addr1 addr2 "user error"
+                                                 threadDelay 100000
```

Comment:
interesting one, but looks scary, lets go with former (node monitoring) approach, because it will fix possible races with d-p internals and latter doesn't.